src/python/pants/option/global_options.py

```diff
@@ -116,6 +125,21 @@ def from_bootstrap_options(cls, bootstrap_options: OptionValueContainer) -> Exec
             token_header = {"authorization": f"Bearer {oauth_token}"}
             remote_execution_headers.update(token_header)
             remote_store_headers.update(token_header)
+        if bootstrap_options.remote_auth_plugin:
+            auth_plugin_path, auth_plugin_func = bootstrap_options.remote_auth_plugin.split(":")
+            auth_plugin_module = importlib.import_module(auth_plugin_path)
+            auth_plugin_func = getattr(auth_plugin_module, auth_plugin_func)
+            auth_plugin_result = cast(
+                AuthPluginResult,
+                auth_plugin_func(
+                    initial_execution_headers=remote_execution_headers,
+                    initial_store_headers=remote_store_headers,
+                ),
```

Comment:
I'm working to add an `options: Options` kwarg so that the plugin can consume other subsystems.